web/static/util.js

```diff
@@ -714,3 +714,57 @@ function _schemaResolveRefs( a_props, a_refs ){
     }
 }
 */
+
+// Note about the below, I am assuming that the client ID is readily available since it is used in datafed-ws.js
+/**
+ * Basic implementation of get_authorize_url from Globus SDK
+ *  @param {UUID} client_id The UUID of the Globus authentication client
+ *  @param {string} redirect_uri The URI safe application-wide Globus Auth redirect URI.
+ *  @param {Array<string>}requested_scopes The scopes on the token(s) being requested
+ *      In the case of accessing a mapped collection, this should include the mapped collection's UUID
+ *      like so: https://auth.globus.org/scopes/YOUR-UUID-HERE/data_access
+ *  @param {string} state Allows the application to pass information back to itself
+ *  @param {boolean} refresh_tokens Request refresh tokens in addition to access tokens
+ *  @param {object} query_params Additional params
+ *  @returns {string} The URL a user can follow to provide authorization and consent via Globus
+ */
+export function globusGetAuthorizeURL(client_id, redirect_uri, requested_scopes=[], state="_default", refresh_tokens=false, query_params={}) {
+    if ( !client_id || !redirect_uri ) {
+        throw new Error("Missing required parameters, please provide client_id and redirect_uri");
+    }
+    if (!requested_scopes?.length) {
+        requested_scopes = [
+            "openid", "profile", "email",
+            "urn:globus:auth:scope:transfer.api.globus.org:all"
+        ];
+    }
+    // TODO: stubbed client object to provide base url, we should extract if possible
+    const auth_client = {
+        base_url: "https://auth.globus.org",
+    };
+    const authorize_base_url = auth_client.base_url + "/v2/oauth2/authorize";

```

Comment:
I mean that it may make sense to have the client abstracted into another class/object like on the reference [ConfidentialAppAuthClient](https://globus-sdk-python.readthedocs.io/en/stable/services/auth.html#globus_sdk.ConfidentialAppAuthClient) and [GlobusAuthorizationCodeFlowManager](https://globus-sdk-python.readthedocs.io/en/stable/services/auth.html#globus_sdk.services.auth.GlobusAuthorizationCodeFlowManager) so we can set up once and use the extracted class both in this PR's code and in what you listed.